base/inference.jl

```diff
@@ -737,6 +752,244 @@ function limit_type_depth(t::ANY, d::Int, cov::Bool, vars::Vector{TypeVar}=TypeV
     return R
 end
 
+function limit_type_size(t::ANY, compare::ANY, source::ANY)
```

Comment:
Could use a comment what this does, especially what the `source` parameter is about. Reading this diff top to bottom, it took me a while to figure it out...